api/numaresourcesoperator/v1/helper/nodegroup/nodegroup.go

```diff
@@ -59,31 +59,44 @@ func (ttr Tree) Clone() Tree {
 // One of the key design assumptions in NROP is the 1:1 mapping between NodeGroups and MCPs.
 // This historical accident should be fixed in future versions.
 func FindTrees(mcps *mcov1.MachineConfigPoolList, nodeGroups []nropv1.NodeGroup) ([]Tree, error) {
+	// node groups are validated by the controller before getting to this phase, so for sure all node groups will be valid at this point.
+	// a valid node group has either PoolName OR MachineConfigPoolSelector, not both. Getting here means operator is deployed on Openshift thus processing MCPs
 	var result []Tree
 	for idx := range nodeGroups {
 		nodeGroup := &nodeGroups[idx] // shortcut
+		treeMCPs := []*mcov1.MachineConfigPool{}
 
-		if nodeGroup.MachineConfigPoolSelector == nil {
-			continue
-		}
-		selector, err := metav1.LabelSelectorAsSelector(nodeGroup.MachineConfigPoolSelector)
-		if err != nil {
-			klog.Errorf("bad node group machine config pool selector %q", nodeGroup.MachineConfigPoolSelector.String())
-			continue
+		if nodeGroup.PoolName != nil {
+			for i := range mcps.Items {
+				mcp := &mcps.Items[i]
+				if mcp.Name == *nodeGroup.PoolName {
+					treeMCPs = append(treeMCPs, mcp)
+					// MCO ensures there are no mcp name duplications
+					break
+				}
+			}
+			// it is valid if no match was found
```

Comment:
@ffromani is this how we want to continue or should we return an error here like 
https://github.com/openshift-kni/numaresources-operator/pull/1018/commits/168e8ed3ea965f71baeece9e2b0b74aa00f1cb3f#diff-31ede3f56b1f873ccf421371942426d8ad0c30160d467d790dd7e050802d25d7R97-R98 ?

Reply:
yes please